helix-term/src/ui/picker.rs

```diff
@@ -344,11 +347,17 @@ impl<T: Item + 'static> Component for FilePicker<T> {
 
 #[derive(PartialEq, Eq, Debug)]
 struct PickerMatch {
-    index: usize,
     score: i64,
+    index: usize,
     len: usize,
 }
 
+impl PickerMatch {
+    fn key(&self) -> impl Ord {
+        (cmp::Reverse(self.score), self.index, self.len)
```

Comment:
```suggestion
        (cmp::Reverse(self.score), self.len, self.index)
```
I believe `len` and `index` must be swapped here.
Otherwise the length tiebreaker would have no effect (as the index is always unique)

Reply:
:thinking: do we need a len at all then? as in, I'd expect

```
aaaaac
bc
```

order perhaps? 

Or maybe not -- I can see the appeal of starting with shorter entitiees... Let's go for a minimal diff for the start

Reply:
Fixed (good thing that rust-analyzer has "flip comma" code action, hehe)

Reply:
Sorting by length is useful because it usually shows you what you want first.
For example `foo/bar` matches `foo/bar.rs` before `foo/bar/baz.rs` this way.
See #4698 and  #4688 for context. `Fzf` shares the same behavior by default and I tested this a bit and it's quite useful in practice.

Note that we only apply this sorting once something is typed into the query.
I think the following example illustrates the intended behaviour.

Consider the initial list of files (sorted alphabetically):

```
aaaaac
ac
bc
```

Once you type `ac` the following is shown:

```
ac
aaaaac
```